src/OpenTelemetry/Trace/TracerProviderBuilder.cs

```diff
@@ -58,45 +59,44 @@ public TracerProviderBuilder SetSampler(Sampler sampler)
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
         public TracerProviderBuilder SetResource(Resource resource)
         {
-            this.Resource = resource ?? Resource.Empty;
+            this.resource = resource ?? Resource.Empty;
             return this;
         }
 
         /// <summary>
         /// Adds given activitysource name to the list of subscribed sources.
         /// </summary>
-        /// <param name="activitySourceName">Activity source name.</param>
+        /// <param name="name">Activity source name.</param>
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-        public TracerProviderBuilder AddActivitySource(string activitySourceName)
-        {
+        public TracerProviderBuilder AddActivitySource(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException($"{nameof(name)} is null or whitespace.");
+            }
+
             // TODO: We need to fix the listening model.
             // Today it ignores version.
-            if (this.ActivitySourceNames == null)
-            {
-                this.ActivitySourceNames = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
-            }
+            this.sources.Add(name);
 
-            this.ActivitySourceNames[activitySourceName] = true;
             return this;
         }
 
         /// <summary>
         /// Adds given activitysource names to the list of subscribed sources.
         /// </summary>
-        /// <param name="activitySourceNames">Activity source names.</param>
+        /// <param name="names">Activity source names.</param>
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-        public TracerProviderBuilder AddActivitySources(IEnumerable<string> activitySourceNames)
-        {
-            // TODO: We need to fix the listening model.
-            // Today it ignores version.
-            if (this.ActivitySourceNames == null)
-            {
-                this.ActivitySourceNames = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+        public TracerProviderBuilder AddActivitySource(IEnumerable<string> names)
```

Comment:
Advantages/disadvantages to make this:

```csharp
public TracerProviderBuilder AddActivitySource(params string[] names)
```

I guess one disadvantage would be that the use of `.ToArray()` may be required.

Reply:
@CodeBlanch mentioned it [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1027#discussion_r467517152).
This is a micro-improvement (together with `AddActivitySource` vs. `AddSource`) that can be addressed in a separate PR.

Reply:
It's too bad C# doesn't let us do `params IEnumerable<string>` eh? Personally in this case I would do `params string[]` because this is initialization code. I feel like users will prefer the convenience and it's unlikely they'll have an `IEnumerable<string>` from some other source in the init phase they'll want to pass us. The 99% case is it's a fixed list set during init. In more general APIs, probably the `IEnumerable<string>` is better.

Reply:
Ah, yes, I thought I had seen this suggestion, and thinking "yea, I like this" 😄. Agreed separate PR makes sense, and also agreed the context of this being initialization code makes `params` seem natural here.